### PR TITLE
Adds new layouts for specific integration scenarios

### LINKED ISSUE #18879: Create new layouts (not available for user selection)
On issue https://github.com/bigbluebutton/bigbluebutton/issues/18878 I am proposing allowing to force a specific layout.

I would like to propose the addition of new layouts, specific for these integrations.

Initially, the following options would probably be enough:

- Cameras only
- Presentation only
- Participants and chat only

As these layouts will only work in integration scenarios, where you open several browsers, I propose that these layouts don't appear in the layout selection window.

### PR DESCRIPTION
**Pull request by vitormateusalmeida**
### What does this PR do?

This PR adds 3 new layouts: "Cameras Only", "Presentation Only", and "Participants and Chat Only", as requested in issue #18879. The layouts are not visible to the user and can only be activated via custom parameters. To improve these layouts, some issues need to be addressed:

* Should the hidden features in each layout remain hidden? For example, in "Cameras Only", should controls such as displaying the user list, opening presentations, sharing screens, etc., be hidden?
* In the "Presentation Only" layout, does external video and screen sharing count as part of the presentation?
* In the "Participants and Chat Only" layout, should the components take up the entire screen?
These questions have not been resolved as they are not part of the scope of the task and are considered as improvements.

### Closes Issue(s)
Closes #18879

### Preview
<details>
<summary>Cameras Only</summary>

image

</details>

<details>
<summary>Presentation Only</summary>

image

</details>

<details>
<summary>Participants and Chat Only</summary>

image

</details>

Addendum:
For this delivery, the controls that have been hidden are using the layout type information to remain hidden. As a suggestion for improvement, it would be beneficial if the layout included information about which controls should be hidden.

### CODE REVIEW

**Comment by TiagoJacobs** [comment]
> Should the hidden features in each layout remain hidden? For example, in "Cameras Only", should controls such as displaying the user list, opening presentations, sharing screens, etc., be hidden?

Yes, controls that will not work (like display user list when layout is cameras only) must be hidden.

> In the "Presentation Only" layout, does external video and screen sharing count as part of the presentation?

Yes, as it refers to the presentation area.


> In the "Participants and Chat Only" layout, should the components take up the entire screen?

Yes, as in the opposite we would have an empty space, which would be very weird. 

---

I see that all these items are needed in order to merge the pr, because if we merge like these, the layouts could be considered broken/with bugs.

Thanks @vitormateusalmeida 



I understand that we should not me

**Comment by vitormateusalmeida** [comment]
@ramonlsouza 
In the "LayoutModalComponent," I added a filter to ignore the added layouts that should not appear in the layout selection modal.
It would be better if there were a flag to identify layouts that should or should not appear in the list. For example, layouts to be used with "enforceLayout" should not appear.
Please check line 110 of the LayoutModalComponent.

File: bigbluebutton-html5/imports/ui/components/layout/layout-manager/camerasOnly.jsx [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    const sidebarSize = sidebarContentWidth.width + sidebarNavWidth.width;
+    const cameraDockBounds = calculatesCameraDockBounds(mediaAreaBounds, sidebarSize);
+    const sidebarContentHeight = calculatesSidebarContentHeight();
+    const mediaBounds = calculatesMediaBounds(
+      mediaAreaBounds,
+      cameraDockBounds,
+      sidebarNavWidth.width,
+      sidebarContentWidth.width,
+      sidebarContentHeight.height,
+    );
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/layout/layout-manager/camerasOnly.jsx:124]: This function expects no arguments

```suggestion
    const mediaBounds = calculatesMediaBounds();
```

File: bigbluebutton-html5/imports/ui/components/layout/layout-manager/participantsChatOnlyLayout.jsx [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    const mediaBounds = calculatesMediaBounds(mediaAreaBounds, sidebarSize);
+    const sidebarContentHeight = calculatesSidebarContentHeight(navbarBounds.height,
+      actionbarBounds.height);
+    const cameraDockBounds = calculatesCameraDockBounds(
+      mediaBounds,
+      mediaAreaBounds,
+      sidebarNavWidth.width,
+      sidebarContentWidth.width,
+      sidebarContentHeight.height,
+    );
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/layout/layout-manager/participantsChatOnlyLayout.jsx:171]: This function expects no arguments

```suggestion
    const cameraDockBounds = calculatesCameraDockBounds();
```

File: bigbluebutton-html5/imports/ui/components/layout/layout-manager/presentationOnlyLayout.jsx [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    const sidebarSize = sidebarContentWidth.width + sidebarNavWidth.width;
+    const mediaBounds = calculatesMediaBounds(mediaAreaBounds, sidebarSize);
+    const sidebarContentHeight = calculatesSidebarContentHeight();
+    const cameraDockBounds = calculatesCameraDockBounds(
+      mediaBounds,
+      mediaAreaBounds,
+      sidebarNavWidth.width,
+      sidebarContentWidth.width,
+      sidebarContentHeight.height,
+    );
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/layout/layout-manager/presentationOnlyLayout.jsx:124]: This function expects no arguments

```suggestion
    const cameraDockBounds = calculatesCameraDockBounds();
```

**Review by ramonlsouza — CHANGES_REQUESTED** [review]
Nice PR! I suggested some changes based on SonarCloud report

**Review by ramonlsouza — APPROVED** [review]

**Pull request merged**